src/Illuminate/Support/Str.php

```diff
@@ -289,6 +289,23 @@ public static function excerpt($text, $phrase = '', $options = [])
         return $start->append($matches[2], $end)->toString();
     }
 
+    /**
+     * @param  string  $text
+     * @param  string|array  $highlights
+     * @param  string  $tag
+     * @return string
+     */
+    public static function highlight($text, $highlights, $tag = 'mark')
+    {
+        if (empty($highlights) || $tag === '') {
+            return $text;
+        }
+
+        $highlights = array_map(fn ($value) => preg_quote($value, '/'), (array) $highlights);
+
+        return preg_replace('/(?<![\w?])('.implode('|', $highlights).')(?![\w?+:.\/*^$()\[{\\}\]|])/', "<$tag>$0</$tag>", $text);
```

Comment:
Why are all these characters in the negative look-ahead?

```\w?+:.\/*^$()\[{\\}\]|```

Reply:
They ensure that for example `Laravel?` doesn't match with `Laravel`, but only with `Laravel?`. Imho this way it's more intuitive use.

Reply:
I agree with your example, but it has the undesired effect (in my opinion) of the following output

```php
Str::highlight('Laravel?', 'Laravel');
// Actual output: Laravel?
// Expected output: <mark>Laravel</mark>?
```

Reply:
> They ensure that for example `Laravel?` doesn't match with `Laravel`, but only with `Laravel?`. Imho this way it's more intuitive use.

Also, i'm pretty sure it wouldn't actually do this, as the `?` is passed through `preg_quote`

Reply:
I think your original implementation was better (and simpler), but just needed to add the `preg_quote`

```php
$highlights = array_map(fn ($value) => preg_quote($value, '/'), (array) $highlights);

return preg_replace('/(?:'.implode('|', $highlights).')/', "<$tag>$0</$tag>", $text);
```